src/shared/ui/charts/TimelineChart/TimelineChart.tsx

```diff
@@ -0,0 +1,125 @@
+import { FC, useMemo } from 'react';
+
+import { format } from 'date-fns';
+import { VictoryAxis, VictoryChart, VictoryScatter } from 'victory-native';
+
+import {
+  colors,
+  DAYS_OF_WEEK_NUMBERS,
+  DAYS_OF_WEEK_SHORT_NAMES,
+  getCurrentWeekDates,
+} from '@shared/lib';
+
+import { Box } from '../..';
+import { ChartItem } from '../types';
+
+type TimelineChartOption = {
+  name: string;
+  value: number;
+};
+
+type ChartRowItem = {
+  optionName: string;
+  timelines: Array<{ date: string; value?: number | null }>;
+};
+
+type Props = {
+  options: Array<TimelineChartOption>;
+  data: Array<ChartItem>;
+};
+
+const TimelineChart: FC<Props> = ({ data, options }) => {
+  const generateXAxisDots = () =>
+    Array.from(Array(7).keys()).map(item => ({ dot: item, value: 0 }));
+
+  const mappedData = useMemo(() => {
```

Comment:
Please add more types into your overall code of this component.
For example not clear what is mappedData